### ISSUE TITLE
Can't sftp and ssh by powershell as Administrators in server2019

### ISSUE THREAD

**Issue by MarkFanBiao** [PowerShell/openssh-portable#439]
There are 3 server machine, one is server2019, others are server2012, the machine in server2019 can't be connected by win_scp, and sftp in the powershell self remind : Connection reset by 127.0.0.1 port 22; And ssh in verbose mode remind that 'debug1: SSH2_MSG_SERVICE_ACCEPT received
Connection reset by 127.0.0.1 port 22' terminated in there, other two machine will perform next line like: 'debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey,password,keyboard-interactive
debug1: Next authentication method: publickey'
I am so confused for the problems, please check the question and explain the confusion, thank you!
please contact us by e-mail: markfanbiao@163.com

**Comment by bagajjal** [comment]
It's very difficult to say the exact problem.
Here are some tips,
1) First enable the file based sshd server logging. By default sshd uses ETW based logging.
Add below lines to %programdata%\ssh\sshd_config,
   SyslogFacility LOCAL0
   LogLevel DEBUG3

Any changes to sshd_config requires sshd service restart.
   net stop sshd
   net start sshd

The sshd server logs are present in %programdata%\ssh\logs\sshd.log

2) Debug using ssh client logs (ssh -vvv user@ip)

3) If key based authentication is failing then verify the permissions of private key.
fyi, default sshd_config looks for public key at  %PROGRAMDATA%/ssh/administrators_authorized_keys

If you are still unable to debug then share the client logs (ssh -vvv user@ip), server side logs (DEBUG3 enabled0 and your scenario.

**Comment by MarkFanBiao** [comment]
> It's very difficult to say the exact problem.
> Here are some tips,
> 
> 1. First enable the file based sshd server logging. By default sshd uses ETW based logging.
>    Add below lines to %programdata%\ssh\sshd_config,
>    SyslogFacility LOCAL0
>    LogLevel DEBUG3
> 
> Any changes to sshd_config requires sshd service restart.
> net stop sshd
> net start sshd
> 
> The sshd server logs are present in %programdata%\ssh\logs\sshd.log
> 
> 1. Debug using ssh client logs (ssh -vvv user@ip)
> 2. If key based authentication is failing then verify the permissions of private key.
>    fyi, default sshd_config looks for public key at  %PROGRAMDATA%/ssh/administrators_authorized_keys
> 
> If you are still unable to debug then share the client logs (ssh -vvv user@ip), server side logs (DEBUG3 enabled0 and your scenario.



> It's very difficult to say the exact problem.
> Here are some tips,
> 
> 1. First enable the file based sshd server logging. By default sshd uses ETW based logging.
>    Add below lines to %programdata%\ssh\sshd_config,
>    SyslogFacility LOCAL0
>    LogLevel DEBUG3
> 
> Any changes to sshd_config requires sshd service restart.
> net stop sshd
> net start sshd
> 
> The sshd server logs are present in %programdata%\ssh\logs\sshd.log
> 
> 1. Debug using ssh client logs (ssh -vvv user@ip)
> 2. If key based authentication is failing then verify the permissions of private key.
>    fyi, default sshd_config looks for public key at  %PROGRAMDATA%/ssh/administrators_authorized_keys
> 
> If you are still unable to debug then share the client logs (ssh -vvv user@ip), server side logs (DEBUG3 enabled0 and your scenario.

OK, Thank you very much for your reply so quickly, i will try the log optional and provide the new infomations, i hope i could solve the problem with the log infomations, thanks again!

**Comment by MarkFanBiao** [comment]
Hello, powerful microsoft  engineer bro, my requirement to use Open-sshd for windows server2019 is to use the sftp transfer files for security. Is there any way to start sftp_server and not by start sshd in advance?
Because in my observation, if one connection establish, there will be one sftp_server.exe and two sshd.exe created in the task manager for windows, and the three process isn't be killed when data is not be transferred in the connection, since I config the 
ClientAliveInterval 15 to anesis the problem!
If there is the way, could you teach me the way? I'm so          grateful thant you reply my question and supply the debug solutions to me.


| |
樊 彪
|
|
fydds521@163.com
|
签名由网易邮箱大师定制
On 4/16/2020 13:26，bagajjal<notifications@github.com> wrote：

It's very difficult to say the exact problem.
Here are some tips,

First enable the file based sshd server logging. By default sshd uses ETW based logging.
Add below lines to %programdata%\ssh\sshd_config,
SyslogFacility LOCAL0
LogLevel DEBUG3

Any changes to sshd_config requires sshd service restart.
net stop sshd
net start sshd

The sshd server logs are present in %programdata%\ssh\logs\sshd.log

Debug using ssh client logs (ssh -vvv user@ip)

If key based authentication is failing then verify the permissions of private key.
fyi, default sshd_config looks for public key at %PROGRAMDATA%/ssh/administrators_authorized_keys

If you are still unable to debug then share the client logs (ssh -vvv user@ip), server side logs (DEBUG3 enabled0 and your scenario.

**Comment by MarkFanBiao** [comment]
Hello, Good bro, I config the sshd_server log and obtain the log infomations for server and client verbose, i filter it for my login with ssh command "ssh -vvv TC_WN_ATT@127.0.0.1", following is the log infos:
server:
84312 2020-04-16 17:22:26.504 debug3: checking match for 'User TC_WN_ATT' user tc_wn_att host 127.0.0.1 addr 127.0.0.1 laddr 127.0.0.1 lport 22
84312 2020-04-16 17:22:26.504 debug1: user tc_wn_att matched 'User TC_WN_ATT' at line 91
84312 2020-04-16 17:22:26.504 debug3: match found
84312 2020-04-16 17:22:26.504 debug3: reprocess config:92 setting ForceCommand internal-sftp
84312 2020-04-16 17:22:26.504 debug3: reprocess config:93 setting ChrootDirectory F:\\IPVA\\SFTPFileData\\Attention
84312 2020-04-16 17:22:26.504 debug3: checking match for 'User TC_WN_HOT' user tc_wn_att host 127.0.0.1 addr 127.0.0.1 laddr 127.0.0.1 lport 22
84312 2020-04-16 17:22:26.504 debug3: match not found
84312 2020-04-16 17:22:26.504 debug3: checking match for 'Group administrators' user tc_wn_att host 127.0.0.1 addr 127.0.0.1 laddr 127.0.0.1 lport 22
84312 2020-04-16 17:22:26.504 debug1: generate_s4u_user_token: LsaLogonUser() failed. User 'tc_wn_att' Status: 0xC0000234 SubStatus 0.
84312 2020-04-16 17:22:26.504 debug3: get_user_token - unable to generate token for user tc_wn_att
84312 2020-04-16 17:22:26.504 debug1: generate_s4u_user_token: LsaLogonUser() failed. User 'tc_wn_att' Status: 0xC0000234 SubStatus 0.
84312 2020-04-16 17:22:26.504 error: get_user_token - unable to generate token on 2nd attempt for user tc_wn_att
84312 2020-04-16 17:22:26.504 fatal: ga_init, unable to resolve user tc_wn_att
84312 2020-04-16 17:22:26.504 debug1: do_cleanup


ssh -vvv:
debug1: Host '127.0.0.1' is known and matches the ECDSA host key.
debug1: Found key in C:\\Users\\appadmin/.ssh/known_hosts:1
debug3: send packet: type 21
debug2: set_newkeys: mode 1
debug1: rekey out after 134217728 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug3: receive packet: type 21
debug1: SSH2_MSG_NEWKEYS received
debug2: set_newkeys: mode 0
debug1: rekey in after 134217728 blocks
debug1: Will attempt key: C:\\Users\\appadmin/.ssh/id_rsa
debug1: Will attempt key: C:\\Users\\appadmin/.ssh/id_dsa
debug1: Will attempt key: C:\\Users\\appadmin/.ssh/id_ecdsa
debug1: Will attempt key: C:\\Users\\appadmin/.ssh/id_ed25519
debug1: Will attempt key: C:\\Users\\appadmin/.ssh/id_xmss
debug2: pubkey_prepare: done
debug3: send packet: type 5
debug3: receive packet: type 7
debug1: SSH2_MSG_EXT_INFO received
debug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521>
debug3: receive packet: type 6
debug2: service_accept: ssh-userauth
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug3: send packet: type 50
debug3: recv - from CB ERROR:10054, io:000002B81A9CE760
Connection reset by 127.0.0.1 port 22


I use the cmd "sftp -P 22 TC_WN_ATT@127.0.0.1" try to login with sftp, and of course, it's failed, the log info in sshd log is:
72608 2020-04-16 17:22:26.488 debug1: Killing privsep child 104544
84312 2020-04-16 17:22:26.504 debug3: receive packet: type 5 [preauth]
84312 2020-04-16 17:22:26.504 debug3: send packet: type 6 [preauth]
84312 2020-04-16 17:22:26.504 debug3: receive packet: type 50 [preauth]
84312 2020-04-16 17:22:26.504 debug1: userauth-request for user TC_WN_ATT service ssh-connection method none [preauth]
84312 2020-04-16 17:22:26.504 debug1: attempt 0 failures 0 [preauth]
84312 2020-04-16 17:22:26.504 debug3: mm_getpwnamallow entering [preauth]
84312 2020-04-16 17:22:26.504 debug3: mm_request_send entering: type 8 [preauth]
84312 2020-04-16 17:22:26.504 debug3: mm_request_receive entering
84312 2020-04-16 17:22:26.504 debug3: monitor_read: checking request 8
84312 2020-04-16 17:22:26.504 debug3: mm_answer_pwnamallow
84312 2020-04-16 17:22:26.504 debug2: parse_server_config: config reprocess config len 542
84312 2020-04-16 17:22:26.504 debug3: checking match for 'User TC_WN_ATT' user tc_wn_att host 127.0.0.1 addr 127.0.0.1 laddr 127.0.0.1 lport 22
84312 2020-04-16 17:22:26.504 debug1: user tc_wn_att matched 'User TC_WN_ATT' at line 91
84312 2020-04-16 17:22:26.504 debug3: match found
84312 2020-04-16 17:22:26.504 debug3: reprocess config:92 setting ForceCommand internal-sftp
84312 2020-04-16 17:22:26.504 debug3: reprocess config:93 setting ChrootDirectory F:\\IPVA\\SFTPFileData\\Attention
84312 2020-04-16 17:22:26.504 debug3: checking match for 'User TC_WN_HOT' user tc_wn_att host 127.0.0.1 addr 127.0.0.1 laddr 127.0.0.1 lport 22
84312 2020-04-16 17:22:26.504 debug3: match not found
84312 2020-04-16 17:22:26.504 debug3: checking match for 'Group administrators' user tc_wn_att host 127.0.0.1 addr 127.0.0.1 laddr 127.0.0.1 lport 22
84312 2020-04-16 17:22:26.504 debug1: generate_s4u_user_token: LsaLogonUser() failed. User 'tc_wn_att' Status: 0xC0000234 SubStatus 0.
84312 2020-04-16 17:22:26.504 debug3: get_user_token - unable to generate token for user tc_wn_att
84312 2020-04-16 17:22:26.504 debug1: generate_s4u_user_token: LsaLogonUser() failed. User 'tc_wn_att' Status: 0xC0000234 SubStatus 0.
84312 2020-04-16 17:22:26.504 error: get_user_token - unable to generate token on 2nd attempt for user tc_wn_att
84312 2020-04-16 17:22:26.504 fatal: ga_init, unable to resolve user tc_wn_att
84312 2020-04-16 17:22:26.504 debug1: do_cleanup


Could you help me to analysis the problem? Good bro, powerful engineer bro, whatever, thank you so much to reply me the log debug way! Best wished!


On 4/16/2020 14:23，樊彪<fydds521@163.com> wrote：
Hello, powerful microsoft  engineer bro, my requirement to use Open-sshd for windows server2019 is to use the sftp transfer files for security. Is there any way to start sftp_server and not by start sshd in advance?
Because in my observation, if one connection establish, there will be one sftp_server.exe and two sshd.exe created in the task manager for windows, and the three process isn't be killed when data is not be transferred in the connection, since I config the 
ClientAliveInterval 15 to anesis the problem!
If there is the way, could you teach me the way? I'm so          grateful thant you reply my question and supply the debug solutions to me.


| |
樊 彪
|
|
fydds521@163.com
|
签名由网易邮箱大师定制
On 4/16/2020 13:26，bagajjal<notifications@github.com> wrote：

It's very difficult to say the exact problem.
Here are some tips,

First enable the file based sshd server logging. By default sshd uses ETW based logging.
Add below lines to %programdata%\ssh\sshd_config,
SyslogFacility LOCAL0
LogLevel DEBUG3

Any changes to sshd_config requires sshd service restart.
net stop sshd
net start sshd

The sshd server logs are present in %programdata%\ssh\logs\sshd.log

Debug using ssh client logs (ssh -vvv user@ip)

If key based authentication is failing then verify the permissions of private key.
fyi, default sshd_config looks for public key at %PROGRAMDATA%/ssh/administrators_authorized_keys

If you are still unable to debug then share the client logs (ssh -vvv user@ip), server side logs (DEBUG3 enabled0 and your scenario.

**Comment by bagajjal** [comment]
There is no way to invoke sftp-server without creating sshd..
sftp data is transferred over ssh connection. 
for each ssh connection we create 2 sshd processes (one as SYSTEM to take care of authentication, other as user to perform user actions).. This is the same architecture followed in the UNIX OpenSSH. This is secure way of doing.

Coming back to your error message..
84312 2020-04-16 17:22:26.504 debug1: generate_s4u_user_token: LsaLogonUser() failed. User 'tc_wn_att' Status: 0xC0000234 SubStatus 0.

ErrorMessage 0xC0000234  means STATUS_ACCOUNT_LOCKED_OUT i.e., The user account has been automatically locked because too many invalid logon attempts or password change attempts have been requested.
-- | --

**Comment by bagajjal** [comment]
We track the issues at https://github.com/PowerShell/win32-openssh

**Comment by MarkFanBiao** [comment]
Ok, thank you! I've got it, more thanks for your explaining of the relationship between sftp_server and sshd.exe, I understand a lot about sftp knowledgement. I checked my login account in computer management and the account TC_WN_ATT is surely locked out, the solutions in internet said it's caused by Local security policy settings limit, what an impressive  course for me.
most thanks to you, thank you microsoft powerful bro, wish you work and enjoy life with good mood!


On 4/17/2020 00:38，bagajjal<notifications@github.com> wrote：

There is no way to invoke sftp-server without creating sshd..
sftp data is transferred over ssh connection.
for each ssh connection we create 2 sshd processes (one as SYSTEM to take care of authentication, other as user to perform user actions).. This is the same architecture followed in the UNIX OpenSSH. This is secure way of doing.

Coming back to your error message..
84312 2020-04-16 17:22:26.504 debug1: generate_s4u_user_token: LsaLogonUser() failed. User 'tc_wn_att' Status: 0xC0000234 SubStatus 0.

ErrorMessage 0xC0000234 means STATUS_ACCOUNT_LOCKED_OUT i.e., The user account has been automatically locked because too many invalid logon attempts or password change attempts have been requested.
-- | --
